server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -117,7 +117,26 @@ public IndexNameExpressionResolver(Settings settings) {
      * indices options in the context don't allow such a case.
      */
     public Index[] concreteIndices(ClusterState state, IndicesOptions options, String... indexExpressions) {
-        Context context = new Context(state, options);
+        Context context = new Context(state, options, false, false);
+        return concreteIndices(context, indexExpressions);
+    }
+
+     /**
+     * Translates the provided index expression into actual concrete indices, properly deduplicated.
+     *
+     * @param state                 the cluster state containing all the data to resolve to expressions to concrete indices
+     * @param options               defines how the aliases or indices need to be resolved to concrete indices
+     * @param resolveToWriteIndex   defines whether to require that aliases resolve to their respective write indices
+     * @param indexExpressions      expressions that can be resolved to alias or index names.
+     * @return the resolved concrete indices based on the cluster state, indices options and index expressions
+     * @throws IndexNotFoundException if one of the index expressions is pointing to a missing index or alias and the
+     * provided indices options in the context don't allow such a case, or if the final result of the indices resolution
+     * contains no indices and the indices options in the context don't allow such a case.
+     * @throws IllegalArgumentException if one of the aliases resolve to multiple indices and the provided
+     * indices options in the context don't allow such a case.
+     */
+    public Index[] concreteIndices(ClusterState state, IndicesOptions options, boolean resolveToWriteIndex, String... indexExpressions) {
```

Comment:
why does this method need to be public ? can we remove it and inline the implementation with it's single caller? it's just two lines.